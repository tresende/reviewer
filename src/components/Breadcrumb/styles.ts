import styled, { css, DefaultTheme } from 'styled-components'

type ItemProps = {
  current?: boolean
}

export const Wrapper = styled.ol`
  display: flex;
  list-style: none;
`

const ItemModifiers = {
  current: (theme: DefaultTheme) => css`
    font-weight: ${theme.font.bold};
    &::after {
      content: '';
    }
  `
}

export const Item = styled.li<ItemProps>`
  ${({ theme, current }) => css`
    font-size: ${theme.font.sizes.xxsmall};
    &::after {
      color: ${theme.colors.gray400};
      margin: 0 0.7rem;
      content: '> ';
    }
    ${current && ItemModifiers.current(theme)};
  `}
`

export const Link = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
    text-decoration: none;
  `}
`
