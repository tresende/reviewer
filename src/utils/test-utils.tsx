import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
