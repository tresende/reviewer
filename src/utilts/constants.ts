const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MAX_RATING = 5
export { SCORE_LABEL, MAX_RATING }
