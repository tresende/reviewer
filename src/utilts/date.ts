export const toPTBRFormat = (date: Date | string) => {
  let toFormat = date
  if (typeof date === 'string') toFormat = new Date(date)
  const formatedDate = new Intl.DateTimeFormat('pt-br').format(toFormat)
  return formatedDate
}
