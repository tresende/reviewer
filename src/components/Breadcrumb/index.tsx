import * as S from './styles'

const Breadcrumb = () => (
  <S.Wrapper>
    <S.Item>
      <a href="#">BeerOrCoffee</a>
    </S.Item>
    <S.Item>
      <a href="#">Coworkings Brasil</a>
    </S.Item>
    <S.Item>
      <a href="#">BrasíS.Itema/DF</a>
    </S.Item>
    <S.Item>Asa Sul</S.Item>
    <S.Item>
      <a href="#">BSB Coworking</a>
    </S.Item>
    <S.Item current>Avaliações</S.Item>
  </S.Wrapper>
)

export default Breadcrumb
