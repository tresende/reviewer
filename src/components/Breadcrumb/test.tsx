import { render } from 'utilts/test-utils'
import Breadcrumb from '.'

describe('<Breadcrumb />', () => {
  it('should render component correctly', () => {
    const { container } = render(<Breadcrumb />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
