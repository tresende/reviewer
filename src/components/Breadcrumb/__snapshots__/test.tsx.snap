// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render component correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

.c1 {
  font-size: 1rem;
  font-weight: 700;
  font-weight: 400;
}

.c1::after {
  color: #CCCCCC;
  margin: 0 0.8rem;
  content: '>';
}

.c2 {
  font-size: 1rem;
  font-weight: 700;
}

<ol
  class="c0"
>
  <li
    class="c1"
  >
    <a
      href="#"
    >
      BeerOrCoffee
    </a>
  </li>
  <li
    class="c1"
  >
    <a
      href="#"
    >
      Coworkings Brasil
    </a>
  </li>
  <li
    class="c1"
  >
    <a
      href="#"
    >
      BrasíS.Itema/DF
    </a>
  </li>
  <li
    class="c1"
  >
    Asa Sul
  </li>
  <li
    class="c1"
  >
    <a
      href="#"
    >
      BSB Coworking
    </a>
  </li>
  <li
    class="c2"
  >
    Avaliações
  </li>
</ol>
`;
